bibliopixel/led/circle.py

```diff
@@ -1,27 +1,26 @@
 from . base import LEDBase
 from .. import data_maker
+from . coord_map import gen_circle, calc_ring_steps, calc_ring_pixel_count
 
 
 class LEDCircle(LEDBase):
 
-    def __init__(self, drivers, rings, maxAngleDiff=0, rotation=0, threadedUpdate=False, masterBrightness=255, maker=data_maker.MAKER):
+    def __init__(self, drivers, rings=None, pixels_per=None,
+                 maxAngleDiff=0, rotation=0,
+                 threadedUpdate=False, masterBrightness=255, maker=data_maker.MAKER):
         super().__init__(drivers, threadedUpdate, masterBrightness, maker)
         self.rings = rings
         self.maxAngleDiff = maxAngleDiff
-        self._full_coords = False
+        full_coords = False
         for r in self.rings:
-            self._full_coords |= (len(r) > 2)
-        self.ringCount = len(self.rings)
-        self.lastRing = self.ringCount - 1
-        self.ringSteps = []
-        num = 0
-        for r in self.rings:
-            if self._full_coords:
-                count = len(r)
-            else:
-                count = (r[1] - r[0] + 1)
-            self.ringSteps.append(360.0 / count)
-            num += count
+            full_coords |= (len(r) > 2)
+
+        if not full_coords:
+            self.rings, self.ringSteps = gen_circle(rings=self.rings, pixels_per=pixels_per)
```

Comment:
a) ring_steps is not new... but fine... ring_pixel_angles... I don't know. It's for each ring, what the angle difference between each pixel is. I suck at names.

b) What would you rather call gen_<object>?